### PR TITLE
Alena burak/tech/2 7/bl question implementation

### PR DESCRIPTION
**Pull request by ElenaBurak**

### CODE REVIEW

File: exalearn-api/ExaLearn.Bl/Interfaces/IQuestionService.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using ExaLearn.Bl.DTO;
+using Shared.Enums;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace ExaLearn.Bl.Interfaces
+{
+    public interface IQuestionService
+    {
+        public Task<QuestionDTO> AddAsync(QuestionDTO question);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Interfaces/IQuestionService.cs:10]: spaces between methods

File: exalearn-api/ExaLearn.Bl/Services/QuestionService.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        public async Task<List<QuestionDTO>> GetByLevelAsync(LevelType level)
+        {
+            var questions = await _questionRepository.GetByLevelAsync(level);
+            return (List<QuestionDTO>)(IEnumerable<QuestionDTO>)_mapper.Map<QuestionDTO>(questions);
+        }
+
+        public async Task<QuestionDTO> SaveAsync(QuestionDTO question)
+        {
+            var q = await _questionRepository.AddAsync(_mapper.Map<Question>(question));
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Services/QuestionService.cs]: rename

File: exalearn-api/ExaLearn.Bl/Services/QuestionService.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        public async Task<List<QuestionDTO>> GetByLevelAndTypeAsync(LevelType level, QuestionType type)
+        {
+            var questions = await _questionRepository.GetByLevelAndTypeAsync(level, type);
+            return (List<QuestionDTO>)(IEnumerable<QuestionDTO>)_mapper.Map<QuestionDTO>(questions);
+        }
+
+        public async Task<List<QuestionDTO>> GetByTypeAsync(QuestionType type)
+        {
+            var questions = await _questionRepository.GetByTypeAsync(type);
+            return (List<QuestionDTO>)(IEnumerable<QuestionDTO>)_mapper.Map<QuestionDTO>(questions);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Services/QuestionService.cs]: remove ()()

File: exalearn-api/ExaLearn.Dal/Interfaces/IAnswerRepository.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using ExaLearn.Dal.Entities;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace ExaLearn.Dal.Interfaces
+{
+    public interface IAnswerRepository : IGenericRepository<Answer>
+    {
+        public Task<List<Answer>> GetByQuestionId(int id);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Interfaces/IAnswerRepository.cs:9]: GetByIdAsync

File: exalearn-api/ExaLearn.Dal/Interfaces/IQuestionRepository.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using ExaLearn.Dal.Entities;
+using Shared.Enums;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace ExaLearn.Dal.Interfaces
+{
+    public interface IQuestionRepository : IGenericRepository<Question>
+    {
+        public Task<List<Question>> GetByLevelAndTypeAsync(LevelType level, QuestionType type);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Interfaces/IQuestionRepository.cs:10]: spaces between methods

File: exalearn-api/ExaLearn.Dal/Repositories/AnswerRepository.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+using System.Threading.Tasks;
+
+namespace ExaLearn.Dal.Repositories
+{
+    public class AnswerRepository : GenericRepository<Answer>, IAnswerRepository
+    {
+        protected new ExaLearnDbContext _appDbContext;
+        public AnswerRepository(ExaLearnDbContext appDbContext) : base(appDbContext)
+        {
+            _appDbContext = appDbContext;
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Repositories/AnswerRepository.cs]: remove context

File: exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+using Shared.Enums;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ExaLearn.Dal.Repositories
+{
+    public class QuestionRepository : GenericRepository<Question>, IQuestionRepository
+    {
+        protected new ExaLearnDbContext _appDbContext;
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs]: remove context

File: exalearn-api/ExaLearn.WebApi/Controllers/QuestionController.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        [HttpPost("save")]
+        public async Task<IActionResult> SaveQuestionAsync([FromBody] QuestionDTO question)
+        {
+            return Ok(await _questionService.SaveAsync(question));
+        }
+
+        [HttpPost("update")]
+        public async Task<IActionResult> UpdateQuestionAsync([FromBody] QuestionDTO question)
```
Comment (romanstat) [inline exalearn-api/ExaLearn.WebApi/Controllers/QuestionController.cs]: remove question

**Review by romanstat — APPROVED** [review]

**Review by artemgael96 — APPROVED** [review]

**Pull request merged**